### PR TITLE
Ensure WRITE_TRUNCATE BigQuery loads to the same table are processed together.

### LINKED ISSUE #23306: [Bug]: BigQueryBatchFileLoads in python loses data when using WRITE_TRUNCATE
### What happened?

When using WRITE_TRUNCATE, if the data being loaded is split into multiple temp tables, the copy job that merges all the results together can end up using WRITE_TRUNCATE for each copy job, resulting in only the last copy job "winning", and overwriting all other jobs.

It looks like there was an attempt to handle this here https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery_file_loads.py#L545, but this code assumes that all inputs to TriggerCopyJobs arrive in the same bundle.  From what I observed this is not the case, looking at the logs from that step, the log lines from that step have different "work" fields for copy job 1 and 2.

There probably needs to be a GBK before this step in order to make sure that all copy jobs actually are executed in the same unit?

### Issue Priority

Priority: 1

### Issue Component

Component: io-py-gcp

### LINKED ISSUE #24535: [Bug]: Bigquery Load jobs with WRITE_TRUNCATE disposition may truncate valid records.
### What happened?

`TriggerCopyJobs` was written to assume all jobs belonging to the same table are processed by the same bundle, but this precondition appears to have been invalidated with future changes to the code. Though this still seems to hold most of the time, it may be violated for large writes, and seems to trigger more on runner v2 than runner v1. 

### Issue Priority

Priority: 1

### Issue Component

Component: io-py-gcp

### PR DESCRIPTION
**Pull request by robertwb**
This fixes #24535 and fixes #23306 in a streaming-update compatible way.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

File: sdks/python/apache_beam/io/gcp/bigquery_file_loads.py [hunk 1]
```diff
@@ -515,9 +515,17 @@ def start_bundle(self):
...
-    destination = element[0]
-    job_reference = element[1]
+  def process(
+      self, element_list, job_name_prefix=None, unused_schema_mod_jobs=None):
+    if isinstance(element_list, tuple):
+      # Allow this for streaming update compatibility while fixing BEAM-24535.
+      self.process_one(element_list)
+    else:
+      for element in element_list:
+        self.process_one(element)
```
Comment (ahmedabu98) [inline sdks/python/apache_beam/io/gcp/bigquery_file_loads.py:525]: Let's pass the side inputs into `process_one`, this is probably causing the failing tests in https://ci-beam.apache.org/job/beam_PreCommit_Python_Phrase/3289/testReport/

**Comment by johnjcasey** [comment]
it looks like proto generation is broken for the python tests

**Comment by ahmedabu98** [comment]
retest this please

**Comment by kennknowles** [comment]
Setting aside GHA (which I don't know if they could have other sources of instability) the scan at https://scans.gradle.com/s/ivqiu46gmopb6 for the Jenkins run says that `:sdks:python:test-suites:tox:py310:testPy310Cloud` failed

**Comment by robertwb** [comment]
Run Python PreCommit

**Comment by robertwb** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/io/gcp/bigquery_file_loads.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/io/gcp/bigquery_file_loads.py]: Yeah, I fixed this, but not it's failing on (unrelated-looking) proto issues. Trying to run tests again...

**Comment by robertwb** [comment]
All the Jenkins tests pass, the github actions issues are unrelated.

**Review by ahmedabu98 — APPROVED** [review]
LGTM

**Pull request merged**